apps/web/src/App.vue

```diff
@@ -158,6 +159,16 @@ export default defineComponent({
         watch(() => vm.$route, (value) => {
             state.isVisible = !state.isEmailVerified && !window.localStorage.getItem('hideNotificationEmailModal') && getRouteAccessLevel(value) >= ACCESS_LEVEL.AUTHENTICATED;
         });
+        watch(() => state.isVisible, async (value) => {
```

Comment:
Suggestion: It would be better to change variable `isVisible` to be more specific :) like- `notificationEmailModalVisible`? or else.